themes/V3/5ePHB/snippets/tableOfContents.gen.js

```diff
@@ -22,37 +22,39 @@ const getTOC = (pages)=>{
 		if(!_.last(_.last(res).children)) add2(null, page);
 		_.last(_.last(res).children).children.push({
 			title    : title,
-			page     : page + 1,
+			page     : page,
 			children : []
 		});
 	};
 
 	const res = [];
-	_.each(pages, (page, pageNum)=>{
-		if(!page.includes("{{frontCover}}") && !page.includes("{{insideCover}}") && !page.includes("{{partCover}}") && !page.includes("{{backCover}}")) {
-			const lines = page.split('\n');
-			_.each(lines, (line)=>{
-				if(_.startsWith(line, '# ')){
-					const title = line.replace('# ', '');
-					add1(title, pageNum);
+	const iframe = document.getElementById('BrewRenderer');
+	const iframeDocument = iframe.contentDocument || iframe.contentWindow.document;
+	const headings = iframeDocument.querySelectorAll('h1, h2, h3');
```

Comment:
This PR has a working method for excluding various headings. Currently, this snippet also always excludes the H4 and H5 headings by skipping them entirely. PR #3254 proposes an alternative method. Having both methods can only lead to confusion and complicate diagnosing author problems.

Recommend we change this to `.querySelectorAll('h1, h2, h3, h4, h5')`  and add `h4, h5, ` to the list of default exclusions in the css. If an author does want h4 or h5 headings in their TOC, they can add overrides to the `brew.style`. 

We should also add a snippet to the Styles snippet menu for "Include H4, H5 in Table of Contents".

Reply:
I added h4, h5, and h6 here. I seem to have missed when in this timeline we added H6, but there you go.

Reply:
Oh, I completely forgot html headings actually go to 6 levels.